### PR TITLE
Noname 135 replace ring implementation

### PR DESCRIPTION
**Pull request by mtszlachetka**
First implementation. Still requires a lot of work and improvements but is much better than what we had before.

### CODE REVIEW

File: backend/src/cryptography/ring_signature.py [hunk 1]
```diff
@@ -1,45 +1,103 @@
...
+        "777DE62AAAB8A8628AC376D282D6ED38",
+        "64E67982428EBC831D14348F6F2F9193",
+        "B5045AF2767164E1DFC967C1FB3F2E55",
+        "A4BD1BFFE83B9C80D052B985D182EA0A",
+        "DB2A3B7313D3FE14C8484B1E052588B9",
+        "B7D2BBD2DF016199ECD06E1557CD0915",
+        "B3353BBB64E0EC377FD028370DF92B52",
+        "C7891428CDC67EB6184B523D1DB246C3",
+        "2F63078490F00EF8D647D148D4795451",
+        "5E2327CFEF98C582664B4C0F6CC41659",
```
Comment (yanvoi) [inline backend/src/cryptography/ring_signature.py:49]: all these should be read from a file, path of which should probably be provided in the configuration

File: backend/src/cryptography/ring_signature.py [hunk 2]
```diff
@@ -1,45 +1,103 @@
...
+
+def verify_lrs(message: str, keys: list[int], signature: list[int]) -> bool:
+
+    concatenated_keys: str = "".join([str(k) for k in keys])
+
+    h = h2(concatenated_keys)
+
+    y0 = signature[0]
+
+    n = int((len(signature) - 1) / 2)
```
Comment (yanvoi) [inline backend/src/cryptography/ring_signature.py:82]: just do:
(len(signature) - 1) // 2

no need to cast to int then

File: frontend/wasm/src/lib.rs [hunk 3]
```diff
@@ -1,15 +1,53 @@
...
+                            777DE62AAAB8A8628AC376D282D6ED38\
+                            64E67982428EBC831D14348F6F2F9193\
+                            B5045AF2767164E1DFC967C1FB3F2E55\
+                            A4BD1BFFE83B9C80D052B985D182EA0A\
+                            DB2A3B7313D3FE14C8484B1E052588B9\
+                            B7D2BBD2DF016199ECD06E1557CD0915\
+                            B3353BBB64E0EC377FD028370DF92B52\
+                            C7891428CDC67EB6184B523D1DB246C3\
+                            2F63078490F00EF8D647D148D4795451\
+                            5E2327CFEF98C582664B4C0F6CC41659";
```
Comment (yanvoi) [inline frontend/wasm/src/lib.rs:49]: let's store these somewhere else if possible, it's not good to hold things like this in source code

**Review by yanvoi — CHANGES_REQUESTED** [review]
great job

these changes required a lot of extensive cryptographic research - let's never forget how much effort was put into it

**Review by yanvoi — APPROVED** [review]
LGTM, and again - good job

**Comment by K4RP4T** [comment]
Several months of extensive research did not go to waste, great job!

**Pull request merged**